client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerWithFilterAndProtobufTest.java

```diff
@@ -48,24 +50,26 @@
 
    @Override
    protected void createCacheManagers() throws Throwable {
-      ConfigurationBuilder cfgBuilder = hotRodCacheConfiguration(getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false));
+      ConfigurationBuilder defaultClusteredCacheConfig = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false);
+      defaultClusteredCacheConfig.encoding().key().mediaType(MediaType.APPLICATION_PROTOSTREAM_TYPE);
+      defaultClusteredCacheConfig.encoding().value().mediaType(MediaType.APPLICATION_PROTOSTREAM_TYPE);
+      ConfigurationBuilder cfgBuilder = hotRodCacheConfiguration(defaultClusteredCacheConfig);
       createHotRodServers(NUM_NODES, cfgBuilder);
       waitForClusterToForm();
 
-      for (int i = 0; i < NUM_NODES; i++) {
-         server(i).addCacheEventFilterFactory("custom-filter-factory", new CustomCacheEventFilterFactory());
-
-         // WARNING! This is not the actual instance used at runtime. A new instance is created instead, so all state is lost unless it is re-created in the constructor!
-         ProtoStreamMarshaller marshaller = new CustomProtoStreamMarshaller();
-
-         server(i).setMarshaller(marshaller);
-      }
-
       //initialize server-side serialization context
       RemoteCache<String, String> metadataCache = client(0).getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
       metadataCache.put("sample_bank_account/bank.proto", Util.getResourceAsString("/sample_bank_account/bank.proto", getClass().getClassLoader()));
       assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
 
+      // The server already has a Protostream <=> Object transcoder, register extra marshallers in the ProtobufMetadataManager.
+      // In standalone servers, those marshallers can be deployed using a jar to have the same effect.
+      for (int i = 0; i < NUM_NODES; i++) {
+         server(i).addCacheEventFilterFactory("custom-filter-factory", new CustomCacheEventFilterFactory());
+         ProtobufMetadataManagerImpl metadataManager = ProtobufMetadataManagerImpl.getProtobufMetadataManager(server(i).getCacheManager());
+         MarshallerRegistration.getMarshallers().forEach(metadataManager::registerMarshaller);
```

Comment:
I would prefer not to make ```ProtobufMetadataManagerImpl.getProtobufMetadataManager(EmbeddedCacheManager cacheManager)``` public.

An alternative is to use          MarshallerRegistration.registerMarshallers(ProtobufMetadataManagerImpl.getSerializationContextInternal(server(i).getCacheManager()));

That's also not going to require adding MarshallerRegistration.getMarshallers() anymore.  Or are we in any trouble because MarshallerRegistration.registerMarshallers also registers the proto files?

Reply:
OK, I will not make it public, it was only a convenience  for tests....

Reply:
No fallout from nucular tests please :)